v1/src/main/java/com/google/cloud/teleport/datadog/DatadogEventPublisher.java

```diff
@@ -181,6 +178,40 @@ String getStringPayload(List<DatadogEvent> events) {
     return GSON.toJson(events);
   }
 
+  static class HttpSendLogsUnsuccessfulResponseHandler implements HttpUnsuccessfulResponseHandler {
+    /*
+      See: https://docs.datadoghq.com/api/latest/logs/#send-logs
+      408: Request Timeout, request should be retried after some time
+      429: Too Many Requests, request should be retried after some time
+    */
+    private static final Set<Integer> RETRYABLE_4XX_CODES = Set.of(408, 429);
+
+    private final Sleeper sleeper = Sleeper.DEFAULT;
+    private final BackOff backOff;
+
+    HttpSendLogsUnsuccessfulResponseHandler(BackOff backOff) {
+      this.backOff = Preconditions.checkNotNull(backOff);
+    }
+
+    @Override
+    public boolean handleResponse(HttpRequest req, HttpResponse res, boolean supportsRetry)
+        throws IOException {
+      if (!supportsRetry) {
+        return false;
+      }
+
+      if (res.getStatusCode() / 100 == 5 || RETRYABLE_4XX_CODES.contains(res.getStatusCode())) {
```

Comment:
nit `res.getStatusCode() / 100 == 5` is a bit unintuitive, might be worthy adding a comment

Reply:
Added some explaining variables instead. 👍